apiserver/facades/client/client/filtering.go

```diff
@@ -114,6 +114,12 @@ func BuildPredicateFor(patterns []string) Predicate {
 				return false, err
 			}
 			return or(shims...)
+		case string: // At this stage, only applicable for application name
```

Comment:
It feels weird that all the others have explicit types, but this one is only a string.

Reply:
This comes from map keys - relations are grouped by application names. Passing anything else in, like []*state.Relation or even worse, an individual relation, will make the code work harder than it needs to... Like we'd need to got through each relation and construct predicates for relation.application name.

The other option that i have considered is not to have a separate section that deals with relation but filter them out as we filter out applications and its units...

I'll change this around and we can consider which implementation suits better.

Reply:
I've removed this all together. 

While testing I've also encovered & fixed issues: (1) where we'd filter out an empty, without units, application even if it'd match a predicate; (2) if application did not match a predicate, we'd return an incorrect flag indicating that we had troubles processing; (3) we were pre-filtering applications and units results based on incorrect assumption.